From edec74f0eebf95456b57be4158385a1a41dc7a4f Mon Sep 17 00:00:00 2001
From: Aswin Murugan <quic_aswinm@quicinc.com>
Date: Fri, 28 Feb 2025 13:18:26 +0530
Subject: [PATCH] PENDING: Fallback action for u-boot env

Added support to set u-boot default env if loading env from
ubootenv partition fails.

Signed-off-by: Aswin Murugan <quic_aswinm@quicinc.com>
Upstream-Status: Pending
---
 env/ufs.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/env/ufs.c b/env/ufs.c
index 11deddc131..efa3cd3d87 100644
--- a/env/ufs.c
+++ b/env/ufs.c
@@ -65,8 +65,10 @@ static int env_ufs_load(void)
 	struct env_ufs_info info;
 	int ret, count;
 
-	if (scsi_get_blk("ubootenv", &info.blk, &info.part))
+	if (scsi_get_blk("ubootenv", &info.blk, &info.part)) {
+		env_set_default("ubootenv partition not found", 0);
 		return -EIO;
+	}
 
 	ALLOC_CACHE_ALIGN_BUFFER(env_t, env, 1);
 
@@ -74,6 +76,7 @@ static int env_ufs_load(void)
 
 	if (blk_dread(info.blk, info.part.start, count, env) != count) {
 		free(env);
+		env_set_default("ubootenv partition read failed", 0);
 		return -EIO;
 	}
 
-- 
2.34.1

